Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutlinerListModel.cpp

```diff
@@ -2426,29 +2426,34 @@ namespace AzToolsFramework
 
     bool EntityOutlinerItemDelegate::editorEvent(QEvent* event, QAbstractItemModel* model, const QStyleOptionViewItem& option, const QModelIndex& index)
     {
-        if (event->type() == QEvent::MouseButtonPress &&
-            (index.column() == EntityOutlinerListModel::Column::ColumnVisibilityToggle || index.column() == EntityOutlinerListModel::Column::ColumnLockToggle))
-        {
-            // Do not propagate click to TreeView if the user clicks the visibility or lock toggles
-            // This prevents selection from changing if a toggle is clicked
-            return true;
-        }
-
         if (event->type() == QEvent::MouseButtonPress)
         {
-            AZ::EntityId entityId(index.data(EntityOutlinerListModel::EntityIdRole).value<AZ::u64>());
-
+            QModelIndex firstColumnIndex = index.siblingAtColumn(EntityOutlinerListModel::ColumnName);
+            AZ::EntityId entityId(firstColumnIndex.data(EntityOutlinerListModel::EntityIdRole).value<AZ::u64>());
             if (auto editorEntityUiInterface = AZ::Interface<EditorEntityUiInterface>::Get(); editorEntityUiInterface != nullptr)
             {
                 auto mouseEvent = static_cast<QMouseEvent*>(event);
 
-                auto entityUiHandler = editorEntityUiInterface->GetHandler(entityId);
+                if (auto entityUiHandler = editorEntityUiInterface->GetHandler(entityId))
+                {
+                    // If lock and visibility can be toggled, outliner clicks should not be propagated to those columns.
+                    bool isToggleColumnActive = (entityUiHandler->CanToggleLockVisibility(entityId) &&
+                            (index.column() == EntityOutlinerListModel::Column::ColumnVisibilityToggle ||
+                                index.column() == EntityOutlinerListModel::Column::ColumnLockToggle));
 
-                if (entityUiHandler && entityUiHandler->OnOutlinerItemClick(mouseEvent->pos(), option, index))
-                {                
-                    return true;
+                    if (!isToggleColumnActive && entityUiHandler->OnOutlinerItemClick(mouseEvent->pos(), option, index))
+                    {
+                        return true;
+                    }
                 }
             }
+
+            if (index.column() == EntityOutlinerListModel::Column::ColumnVisibilityToggle || index.column() == EntityOutlinerListModel::Column::ColumnLockToggle)
+            {
+                // Do not propagate click to TreeView if the user clicks the visibility or lock toggles
+                // This prevents selection from changing if a toggle is clicked
+                return true;
+            }
```

Comment:
This check should still be as close to the top as possible to early-out, so right inside the `if (event->type() == QEvent::MouseButtonPress)` line